### PR TITLE
Refine Video::Autoplay options

### PR DESCRIPTION
**Pull request by peak3d**
## Description
see title

## Motivation and Context
Users want to have more control in regards of Autoplay.
For example Episodes should be continued, Movies not.

## How Has This Been Tested?
Win10 / amazon addon / enabled Episodes in Settings and:
- played Movise -> No Autoplay
- played Episode -> Autoplay works

## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/settings/SettingsBase.cpp [hunk 1]
```diff
@@ -175,6 +175,14 @@ void CSettingsBase::UnregisterCallback(ISettingCallback* callback)
   m_settingsManager->UnregisterCallback(callback);
 }
 
+bool CSettingsBase::FindIntInList(const std::string &id, int value) const
+{
+  if (id.empty())
+    return nullptr;
```
Comment (Rechi) [inline xbmc/settings/SettingsBase.cpp]: returntype of the function is bool

**Comment by da-anda** [comment]
not sure if it's a good idea to have 0 as a valid value. Have you checked what happens on a migration from Krypton to Leia and the user thas this setting not turned on (which might result in a 0 value when converted to an int)?

Would it probably be better/easier if we keep the boolean on/off switch and add an additional "limit autoplay to these media types" setting (indented, so with the boolean setting as parent setting). Just an idea for discussion - no demand to change anything.

**Comment by peak3d** [comment]
@da-anda 
- I tested with the booleans true and false and it does not make any difference, everything is disabled.
- I thought about the parent setting too, but didn't see any advantage. The only thing you do is switching the limit list on and off, makes sense for component logging (you modify this often) but IMO not necessary for the autoplaynext functionality (which you set up not very often)

File: xbmc/settings/SettingsBase.cpp (hunk 1 above)
Comment (peak3d) [reply xbmc/settings/SettingsBase.cpp]: thx

**Comment by peak3d** [comment]
I finally need a proper name for the unclassiied streams.
Currently they are titled "Others" but could also be "Unclassified" or "Unknown"

Suggestions appreciated, thx!

**Comment by scott967** [comment]
Since this PR was merged I realized that there is now a settings option for video autoplay on musicvideos.  However, when content is musicvideos or recently added musicvideos, the setting for Music autoplay is used, not video autoplay.  The settings options for musicvideos under video autoplay should be removed.

**Comment by enen92** [comment]
Added wiki-needed as it touches the settings layout. Ref: https://kodi.wiki/view/Settings/Videos

**Comment by Dnkhatri** [comment]
This patch stops external subs from working auto play next is selected for Episodes.

**Comment by KarellenX** [comment]
Wiki done... https://kodi.wiki/view/Settings/Player/Videos#Play_next_video_automatically

**Pull request merged**